subsys/fs/nffs_fs.c

```diff
@@ -0,0 +1,534 @@
+/*
+ * Copyright (c) 2017 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <zephyr/types.h>
+#include <errno.h>
+#include <init.h>
+#include <flash.h>
+#include <fs.h>
+#include <crc16.h>
+#include <misc/__assert.h>
+#include <misc/printk.h>
+#include <nffs/nffs.h>
+#include <nffs/os.h>
+
+static struct k_mutex nffs_lock;
```

Comment:
Can you add a comment here elaborating on the need for this mutex? In particular how the NFFS code is not thread safe and requires locking outside of its calls

Reply:
thanks!